string location
---
int32 result
